src/main/java/core/basesyntax/service/Buy.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitBox;
+import java.util.Map;
+
+public class Buy extends Operation {
+    public Buy(OperationType type, FruitBox fruit) {
+        super(type, fruit);
+    }
+
+    @Override
+    public Map<String, FruitBox> changeQuantity(int totalAmount,
+                                                Map<String, FruitBox> storage) {
+        String fruitName = fruitBox.getFruitType();
```

Comment:
let's make storage — this class field and not pass it to this method

Reply:
done